testsuite/cluster-test/src/tx_emitter.rs

```diff
@@ -678,15 +727,33 @@ impl SubmissionWorker {
             .iter_mut()
             .choose_multiple(&mut rng, batch_size);
         let mut requests = Vec::with_capacity(accounts.len());
+        let mut invalid_size = if self.invalid_tx != 0 {
+            // if enable mix invalid tx, at least 1 invalid tx per batch
+            max(1, accounts.len() * self.invalid_tx as usize / 100)
+        } else {
+            0
+        };
         for sender in accounts {
             let receiver = self
                 .all_addresses
                 .choose(&mut rng)
                 .expect("all_addresses can't be empty");
-            let request =
-                gen_transfer_txn_request(sender, receiver, SEND_AMOUNT, self.chain_id, gas_price);
-            requests.push(request);
+            if invalid_size > 0 {
```

Comment:
This would batch the bad txns together. Randomly mixing/interleaving would have been better.  Anyhow, I think this is good enough to get started.

Reply:
I think u are correct, I can do a shuffle after we initialized all requests.